build/Microsoft.DotNet.Cli.Publish.targets

```diff
@@ -0,0 +1,107 @@
+  <UsingTask TaskName="FinalizeBuildTask" AssemblyFile="$(CLIBuildDll)" />
+  <UsingTask TaskName="SetBlobPropertiesBasedOnFileTypeTask" AssemblyFile="$(CLIBuildDll)" />
+
+  <!-- PUBLISH_TO_AZURE_BLOB env variable set by CI -->
+  <Target Name="Publish"
+          Condition=" '$(PUBLISH_TO_AZURE_BLOB)' != '' "
+          DependsOnTargets="Prepare;
+                            Package;
+                            PublishArtifacts;
+                            FinalizeBuild" />
+
+  <Target Name="PublishArtifacts" DependsOnTargets="SetupAzureBlobInformation;
+                                                    GenerateVersionBadge;
+                                                    GatherItemsForPattern;
+                                                    UploadToAzure;
+                                                    PublishDebFilesToDebianRepo;
+                                                    PublishCliVersionBadge" />
+
+  <Target Name="FinalizeBuild">
+    <FinalizeBuildTask AccountName="$(CloudDropAccountName)"
+                       AccountKey="$(CloudDropAccessToken)"
+                       NugetVersion="$(NugetVersion)"
+                       Channel="$(Channel)"
+                       CommitHash="$(CommitHash)"
+                       BranchName="$(BranchName)" />
+  </Target>
+
+  <Target Name="SetupAzureBlobInformation">
+    <PropertyGroup>
+      <Product>Sdk</Product>
+      <ContainerName>$(STORAGE_CONTAINER)</ContainerName>
+      <CloudDropAccessToken>$(STORAGE_KEY)</CloudDropAccessToken>
+      <CloudDropAccountName>$(STORAGE_ACCOUNT)</CloudDropAccountName>
+      <DotnetBlobRootUrl>https://$(CloudDropAccountName).blob.core.windows.net/$(ContainerName)</DotnetBlobRootUrl>
+    </PropertyGroup>    
+  </Target>
+
+  <Target Name="GatherItemsForPattern">
+    <ItemGroup>
+      <ForPublishing Include="@(GeneratedInstallers)" />
+      <ForPublishing Include="%(GenerateArchivesInputsOutputs.Outputs)" />
+    </ItemGroup>
+
+    <ItemGroup>
+      <ForPublishing>
+        <RelativeBlobPath>$(Product)/$(NugetVersion)/$([System.String]::Copy('%(Filename)%(Extension)').Replace('\' ,'/'))</RelativeBlobPath> 
+      </ForPublishing>
+    </ItemGroup>
+  </Target>  
+
+  <Target Name="PublishDebFilesToDebianRepo" Condition=" '$(OSName)' == 'ubuntu' ">
+    <Error Condition="'$(REPO_ID)' == ''" Text="REPO_ID must be set as an environment variable for debian publishing." />
+
+    <ItemGroup>
+      <SdkInstallerFileItemGroup Include="$(SdkInstallerFile)" />
+    </ItemGroup>
+    <PropertyGroup>
+      <SdkDebianPackageName>dotnet-dev-$(NugetVersion)</SdkDebianPackageName>
+      <SdkDebianUploadUrl>$(DotnetBlobRootUrl)/$(Product)/$(NugetVersion)/%(SdkInstallerFileItemGroup.Filename)%(SdkInstallerFileItemGroup.Extension)</SdkDebianUploadUrl>
+      <DebianUploadJsonFile>$(PackageDirectory)/package_upload.json</DebianUploadJsonFile>
```

Comment:
This file should go somewhere in intermediates ideally. `SdkDebianIntermediateDirectory` perhaps?